### PR TITLE
Fix SDP mangling in DTMF and add a test

### LINKED ISSUE #752: Make sure the DTMF sample removes all non desired codecs and all other references to it in the SDP
Check https://code.google.com/p/chromium/issues/detail?id=585181 for context.

### PR DESCRIPTION
**Pull request by KaptenJansson**
**Description**
Update the SDP mangling code.

Fixes #752 

**Purpose**
Fix the DTMF demo

One curious thing though, testing on Stable, Beta and Dev yields 3 different results.
- Stable (49.0.2623.87) - Works as intended
- Beta (50.0.2661.37) - Works but crackling noise
- Dev (51.0.2679.0) - Does not work at all, audioInputLevel is non zero but it does not actually send the audio (bitrate is 0).

Will continue investigating and commit to this PR.

CC @tednakamura

### CODE REVIEW

**Comment by KaptenJansson** [comment]
Turns out the SDP manging code is no longer needed ;).

The issues still remain however they seem to be unrelated to DTMF as I can reproduce them on other samples.

**Comment by KaptenJansson** [comment]
OK so the choppy audio on M50 was temporary, after a crash when trying to create an aecdump it now sounds OK. Still investigating the M51 issue, however as mentioned it's unrelated to this change.

**Comment by KaptenJansson** [comment]
Interesting, it works fine in the tests on all Chrome channels....Maybe some local issue related to my profile....

**Comment by KaptenJansson** [comment]
@samdutton PTAL when you get a chance.

**Comment by KaptenJansson** [comment]
PTAL @hkjellander

File: src/content/peerconnection/dtmf/js/test.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+ *  Copyright (c) 2016 The WebRTC project authors. All Rights Reserved.
+ *
+ *  Use of this source code is governed by a BSD-style license
+ *  that can be found in the LICENSE file in the root of the source
+ *  tree.
+ */
+ /* eslint-env node */
+
+'use strict';
+// This is a basic test file for use with testling.
```
Comment (kjellander-unity) [inline src/content/peerconnection/dtmf/js/test.js:11]: testling -> testing.

File: src/content/peerconnection/dtmf/js/test.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+      return driver.executeScript(
+          'return document.querySelector(\'#dtmfStatus\').innerHTML === ' +
+          '\'DTMF activated\';');
+    }, 30 * 1000);
+  })
+  .then(function() {
+    t.pass('DTMF activated');
+    // Set the tones to be sent.
+    return driver.executeScript(
+        'document.querySelector(\'#tones\').value = \'1 # , 9 \'');
```
Comment (kjellander-unity) [inline src/content/peerconnection/dtmf/js/test.js:43]: Why the whitespace between the digits, pound and comma?

File: src/content/peerconnection/dtmf/js/test.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+          'return document.querySelector(' +
+          '\'#sentTones\').innerHTML.length === 9');
+    });
+  }, 30 * 1000)
+  .then(function() {
+    return driver.executeScript(
+        'return document.querySelector(\'#sentTones\').innerHTML');
+  })
+  .then(function(sentTones) {
+    t.ok(sentTones === '1 # , 9  ', 'sentTones matches tones');
```
Comment (kjellander-unity) [inline src/content/peerconnection/dtmf/js/test.js:62]: Here there's an additional whitespace after the 9. What's the purpose of that?

**Comment by kjellander-unity** [comment]
Is there an easy way I can try out the fixed page or do I have to patch a fork and use that?

File: src/content/peerconnection/dtmf/js/test.js (hunk 1 above)
Comment (fippo) [inline src/content/peerconnection/dtmf/js/test.js:11]: i think this was testling but we moved away from testling to selenium

Comment (fippo) [inline src/content/peerconnection/dtmf/js/test.js:11]: @KaptenJansson should we delete that line everywhere?

Comment (KaptenJansson) [inline src/content/peerconnection/dtmf/js/test.js:11]: Yes that should be removed.

File: src/content/peerconnection/dtmf/js/test.js (hunk 2 above)
Comment (KaptenJansson) [inline src/content/peerconnection/dtmf/js/test.js:43]: That's what I get when sending the tones, I'm just matching the output like I wrote [here](https://github.com/webrtc/samples/pull/767/files/5075286359ee1a59bfdec47473387e007ce0ee44#diff-821d966122db351688988c1a9a2358acR51)

File: src/content/peerconnection/dtmf/js/test.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    return driver.executeScript(
+        'document.querySelector(\'#tones\').value = \'1 # , 9 \'');
+  })
+  .then(function() {
+    return driver.findElement(webdriver.By.id('sendTonesButton')).click();
+  })
+  .then(function() {
+    return driver.wait(function() {
+      return driver.executeScript(
+        // For some reason the demo sends and extra space. Assume its due to
```
Comment (kjellander-unity) [inline src/content/peerconnection/dtmf/js/test.js]: its -> it's

File: src/content/peerconnection/dtmf/js/test.js (hunk 2 above)
Comment (kjellander-unity) [inline src/content/peerconnection/dtmf/js/test.js:43]: OK

File: src/content/peerconnection/dtmf/js/test.js (hunk 3 above)
Comment (KaptenJansson) [inline src/content/peerconnection/dtmf/js/test.js:62]: same as the comment above.

File: src/content/peerconnection/dtmf/js/test.js (hunk 4 above)
Comment (KaptenJansson) [inline src/content/peerconnection/dtmf/js/test.js]: Done

**Comment by KaptenJansson** [comment]
fork (assuming nodejs is installed) and run ´node test/server.js´ then browse to ´https://localhost:8080/´

**Comment by kjellander-unity** [comment]
LGTM, I assume you'll fix the testling references for all tests in another PR (I wasn't aware it was the name of a test framework ;)

**Pull request merged**